cpp/cmake/thirdparty/patches/thrust_override.json

```diff
@@ -32,6 +32,11 @@
           "file" : "${current_json_dir}/cub_segmented_sort_with_bool_key.diff",
           "issue" : "Fix an error in CUB DeviceSegmentedSort when the keys are bool type [https://github.com/NVIDIA/cub/issues/594]",
           "fixed_in" : "2.1"
+        },
+        {
```

Comment:
This shouldn't be in the Thrust patches override file as it will be applied when Thrust is downloaded and not when nvbench is.

We will also want to make a second file ( nvbench_override ) and have just the new entry in it:

```
{
  "packages" : {
    "nvbench" : {
      "patches" : [        
        {
          "file" : "${current_json_dir}/nvbench_global_setup.diff",
          "issue" : "Fix add support for global setup to initialize RMM in nvbench [https://github.com/NVIDIA/nvbench/pull/123]",
          "fixed_in" : ""
        }
      ]
    }
  }
}
```

We will want to add a `get_nvbench.cmake` and have it do the following logic before call `rapids_cpm_nvbench`:
```cmake
  include(${rapids-cmake-dir}/cpm/nvbench.cmake)
  include(${rapids-cmake-dir}/cpm/package_override.cmake)

  set(cudf_patch_dir "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/patches")
  rapids_cpm_package_override("${cudf_patch_dir}/nvbench_override.json")
```